src/app/course-page/services/item-course.service.ts

```diff
@@ -1,96 +1,73 @@
 import { Injectable } from '@angular/core';
 import { CourseItem } from '../models/course-item.model';
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Observable, throwError, of } from 'rxjs';
+import {catchError, map, debounceTime, tap, distinctUntilChanged } from 'rxjs/operators';
 
 @Injectable({
   providedIn: 'root'
 })
 export class ItemCourseService {
   public currentId: number;
-  public items: CourseItem[] = [
-    {
-      id: 1,
-      title: 'Video converting course',
-      description: 'Learn about where you can find course descriptions, what information they include, how they work, and details about various components of a course description. Course descriptions report information about a university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course schedules that contain descriptions for all courses offered during a particular semester.',
-      duration: 130,
-      dateObj: Date.parse("January 26, 2020"),
-      topRated: false
-    },
-    {
-      id: 2,
-      title: 'React in other life course',
-      description: 'some text here',
-      duration: 40,
-      dateObj: Date.parse("Nov 12, 2019"),
-      topRated: false
-    },
-    {
-      id: 3,
-      title: 'Awesome course', description: 'fdf gsdsf dgfdfgdf dhdghrqertqrt tretw',
-      duration: 95,
-      dateObj: Date.parse("Feb 20, 2020"),
-      topRated: false
-    },
-    {
-      id: 4,
-      title: 'Angular course',
-      description: 'fggdfg dgfdfg gdfgd lorem',
-      duration: 56,
-      dateObj: Date.parse("Nov 3, 2019"),
-      topRated: false
-    },
-    {
-      id: 5,
-      title: 'Javascript language course',
-      description: 'fggdfg dgfdfg gdfgd lorem',
-      duration: 67,
-      dateObj: Date.parse("Nov 11, 2019"),
-      topRated: true
-    },
-    {
-      id: 6,
-      title: 'Some more else course',
-      description: 'fggdfg dgfdfg gdfgd lorem',
-      duration: 202,
-      dateObj: Date.parse("Oct 30, 2019"),
-      topRated: true
-    }
-  ]
+  public items: CourseItem[] = [];
+
+  public start = '0';
+  public count = '5';
 
-  constructor() { }
+  constructor(private http: HttpClient) { }
 
   // Get list
-  public getItems(): CourseItem[] {
-    return this.items;
+  getItems(): Observable<CourseItem[]> {
+    let params = new HttpParams();
```

Comment:
I suggest using the following  approach: this.http.get('http://localhost:888.....', { headers: headers, search: params })
Just pass parameters with their names and dont create separate new HttpParams.

Reply:
set params like without new HttpParams
      params: {
        'start': startIndex,
        'count': count
      }